core/src/main/java/com/yahoo/oak/SeqExpandMemoryManager.java

```diff
@@ -111,31 +70,311 @@ public long getInvalidReference() {
 
     @Override
     public boolean isReferenceValid(long reference) {
-        return rcd.isReferenceValid(reference);
+        return rc.isReferenceValid(reference);
     }
 
     @Override
     public boolean isReferenceDeleted(long reference) {
-        return rcd.isReferenceDeleted(reference);
+        return rc.isReferenceDeleted(reference);
     }
 
     @Override
     public boolean isReferenceValidAndNotDeleted(long reference) {
-        return isReferenceValid(reference);
+        return rc.isReferenceValid(reference);
     }
 
     @Override
     public boolean isReferenceConsistent(long reference) {
-        return rcd.isReferenceConsistent(reference);
+        return rc.isReferenceConsistent(reference);
     }
```

Comment:
Specifically here it is going to be wrapping anyway, because `ReferenceCodec` in either way is internal to any MM.

Regarding removing all the child implementation of `ReferenceCoded` I am not sure about it. `ReferenceCodecSeqExpand` can be removed as it indeed doesn't have any logic. But `ReferenceCodecSyncRecycle` does have internal logic...